client/modules/sheaf/src/PaneEditorTopbar.svelte

```diff
@@ -0,0 +1,172 @@
+<!--
+  @component Sheaf Editor: Editor Pane Topbar.
+-->
+<script lang="ts">
+  import { getContext } from "svelte"
+  import type { SheafContext } from "./context"
+  import { Button } from "components"
+  import { throttle } from "wj-util"
+  import { t, number, unit } from "wj-state"
+
+  const { editor, bindings, settings, small } = getContext<SheafContext>("sheaf")
+
+  // -- STATS
+
+  let chars = 0
+  let bytes = 0
+  let words = 0
+  let lines = 0
+
+  function kilobyteCount(str: string) {
+    // this is a really _spicy_ way of getting a rough kilobyte count
+    // it doesn't need to be precise at all, so this is fine.
+    return Math.round((encodeURI(str).split(/&..|./).length - 1) / 1000)
+  }
+
+  function wordCount(str: string) {
+    return str.trim().split(/\s+/).length
+  }
```

Comment:
That's not really the point though, it's just a general count of number of space-separated chunks in the work.

Reply:
although I totally didn't think about this.

... I do have access to an AST or token stream................ yes.... very accurate word count....

I'll actually think about solving the word count properly, as I'm sure it'll get used as a point of comparison and I'd like for it to be close to correct.

Reply:
Alright I've thought about and I this isn't so bad to do, because I have access to the editors syntax tree, and there are efficient methods to iterate across the entire tree. All I'd need to do for a 95% accurate word count is go through every `content` leaf and just sum up the word counts. 

This won't catch any text you put inside, say, the argument of a collapsible, but as long as you're not a nutjob and put most of your article in arguments it should be fine. This will work for includes btw